### DISCUSSION TITLE
Problem when including QTF for high sea states

### DISCUSSION THREAD

**Discussion by thanasisder** [OpenFAST/openfast#1985] [General]
I have a spar and I want to study the slow-drift motions in a quite severe sea state (Hs=15.6 m, Tp = 15.2 s). I compare OpenFAST while having linear hydrodynamics, Newmann approximation and QTF (from Hydrostar). Also to mention that I use input wave elevation (method 5).
When I include the QTF, I receive this error at an instant in the simulation:

Would you have any idea what causes the problem? 

----------------------------
 FAST_Solution:FAST_AdvanceStates:ED_ABM4:ED_CalcContStateDeriv:SetCoordSy:Small angle assumption
 violated in SUBROUTINE SmllRotTrans() due to a large platform displacement (ElastoDyn
 SetCoordSy). The solution may be inaccurate. Simulation continuing, but future warnings from
 SmllRotTrans() will be suppressed.
  Additional debugging message from SUBROUTINE SmllRotTrans(): 782.27 s

 FAST_Solution:CalcOutputs_And_SolveForInputs:SolveOption1:ED_HD_InputOutputSolve:HydroDyn_CalcOutp
 ut: Angles in GetSmllRotAngs() are larger than 0.4 radians.
 HydroDyn_CalcOutput:HDOut_MapOutputs: Angles in GetSmllRotAngs() are larger than 0.4 radians.
 ED_HD_InputOutputSolve:HydroDyn_CalcOutput: Angles in GetSmllRotAngs() are larger than 0.4
 radians.
 HydroDyn_CalcOutput:HDOut_MapOutputs: Angles in GetSmllRotAngs() are larger than 0.4 radians.

 FAST_Solution:CalcOutputs_And_SolveForInputs:SolveOption1:ED_HD_InputOutputSolve:HydroDyn_CalcOutp
 ut: Angles in GetSmllRotAngs() are larger than 0.4 radians.
 HydroDyn_CalcOutput:HDOut_MapOutputs: Angles in GetSmllRotAngs() are larger than 0.4 radians.
 ED_HD_InputOutputSolve:HydroDyn_CalcOutput: Angles in GetSmllRotAngs() are larger than 0.4
 radians.
 HydroDyn_CalcOutput:HDOut_MapOutputs: Angles in GetSmllRotAngs() are larger than 0.4 radians.

  MD_UpdateStates: NaN state detected.

 FAST_Solution:FAST_AdvanceStates: MD_UpdateStates: NaN state detected.

 OpenFAST encountered an error at simulation time 782.38 of 7900 seconds.
  Simulation error level: FATAL ERROR

  Aborting OpenFAST.

----------------------------

**Comment by jjonkman** [comment]
Dear @thanasisder,

I guess my first question is whether the spar you are simulating is designed to survive a sea state of that severity.

The next question is whether the warnings and error are triggered by a physical issue (large floater motion caused by the severe sea state) or a numerical issue.  To figure out the latter, you could share plots of the floater motion (surge, sway, heave, roll, pitch, yaw) and associated wave elevation to see if the large floater rotation at approx. 782 s is a rare anomaly (indicating a potential numerical issue) or simply the result of a large wave.

Best regards,

**Reply by thanasisder** [reply]
Hello Dr. Jonkman and thank you for your prompt response!
1) The sea state was studied experimentally without any problem so I believe that yes.
2) Regarding the plots I initially saw that it was producing some unphysically large motions in roll and yaw (very large peaks at the crashing time), so I deactivated the QTF in all the non-horizontal plane DOFs, since we have unidirectional waves. It still crashes at a later instant in time due to an unphysical pitch motion this time as you can see at the plots below 

![image](https://github.com/OpenFAST/openfast/assets/62059192/6cd22c26-ce4c-409a-bafd-f5c8bc388711)

**Reply by jjonkman** [reply]
Dear @thanasisder,

 Are you receiving the same warnings and error for this updated case with horizontal plane QTF?

Can you also share corresponding plots of sway, roll, pitch, and wave elevation?

Best regards,

**Reply by thanasisder** [reply]
Dear Dr. Jonkman,
Yes the same exact warnings.  
Here are the corresponding plots. Apparently the problem in sway, roll and yaw persists. 

![image](https://github.com/OpenFAST/openfast/assets/62059192/a5e2df61-bb85-4723-a923-0d0a1abb4d14)

Best regards,
Thanasis

**Reply by jjonkman** [reply]
Dear @thanasisder,

It appears to me that the deep trough in the wave is triggering some an instability in the model, resulting in large floater motions (especially yaw) that crash MoorDyn.   Do changes to basic solver settings in the model such as reducing `DT`, reducing `DT_UJac`, or increasing `NumCrctn` in the OpenFAST primary (_*.fst_) input file change the solution through this severe wave event?

Best regards,

**Reply by thanasisder** [reply]
Hello Dr. Jonkman,
I will investigate the effect of those parameters and come back at you as soon as I have some answer to your question!
Best regards,
Thanasis

**Comment by thanasisder** [comment]
Hello again,
I reduced the DT_Ujac from 9999 to 10xtimestep and it runs perfectly now! Thank you very much for your help, you can consider this issue as closed!

Best regards,
Thanasis
